web/src/components/navbar/desktop/NavBarDesktop.tsx

```diff
@@ -0,0 +1,185 @@
+import { ButtonIcon } from "@/components/ButtonIcon";
+import { NavBarLogoOnly } from "@/components/navbar/NavBarLogoOnly";
+import { DesktopLoginButton } from "@/components/navbar/desktop/DesktopLoginButton";
+import { NavBarDesktopDropDown } from "@/components/navbar/desktop/NavBarDesktopDropDown";
+import { NavBarDesktopHomeLogo } from "@/components/navbar/desktop/NavBarDesktopHomeLogo";
+import { NavBarDesktopQuickLinks } from "@/components/navbar/desktop/NavBarDesktopQuickLinks";
+import { NavBarVerticalLineDivider } from "@/components/navbar/desktop/NavBarDesktopVerticalLineDivider";
+import { NavBarDesktopWrapper } from "@/components/navbar/desktop/NavBarDesktopWrapper";
+import {
+  AddBusinessItem,
+  GetStartedMenuItem,
+  LogoutMenuItem,
+  MyNjMenuItem,
+  ProfileMenuItem,
+  RegisterMenuItem,
+} from "@/components/navbar/shared-submenu-components";
+import { Icon } from "@/components/njwds/Icon";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { getBusinessIconColor } from "@/lib/domain-logic/getBusinessIconColor";
+import { getNavBarBusinessTitle } from "@/lib/domain-logic/getNavBarBusinessTitle";
+import { orderBusinessIdsByDateCreated } from "@/lib/domain-logic/orderBusinessIdsByDateCreated";
+import React, { ReactElement, useEffect, useRef, useState } from "react";
+
+interface Props {
+  previousBusinessId?: string | undefined;
+  isLanding: boolean | undefined;
+  logoOnlyType?: "NAVIGATOR_LOGO" | "NAVIGATOR_MYNJ_LOGO" | undefined;
+  currentlyOnboarding: boolean | undefined;
+  isAuthenticated: boolean;
+}
+
+export const NavBarDesktop = (props: Props): ReactElement => {
+  const { business, userData } = useUserData();
+  const { Config } = useConfig();
+
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef<HTMLButtonElement | null>(null);
+
+  const handleClose = (
+    event?: MouseEvent | TouchEvent | React.MouseEvent<HTMLLIElement> | React.MouseEvent<Document>
+  ): void => {
+    if (event && anchorRef.current && anchorRef.current.contains(event.target as Node)) {
+      return;
+    }
+
+    setOpen(false);
+  };
+
+  const prevOpen = useRef(open);
+  useEffect(() => {
+    if (null !== anchorRef.current && prevOpen.current && !open) {
+      anchorRef.current.focus();
+    }
+    prevOpen.current = open;
+  }, [open]);
+
+  const textColor = props.isAuthenticated ? "primary" : "base";
+  const navBarBusinessTitle = getNavBarBusinessTitle(business, props.isAuthenticated);
+  const currentIndex =
+    userData && business ? orderBusinessIdsByDateCreated(userData).indexOf(business.id) : 0;
+
+  if (props.logoOnlyType) {
+    // loading/redirect/ethan onboarding
+    return <NavBarLogoOnly logoType={props.logoOnlyType} />;
+  } else if (props.isLanding) {
+    // landing
```

Comment:
pebble - to avoid using these comments you could extract each navbar component into it's own component that lives within the page 

`const LandingNavBar = ({ ...props }) => {/* ... */};`
`const OnboardingNavBar = ({ ...props }) => {/* ... */};`

Reply:
I thought about that and elected not to. 

The reason is because I think that any change you might make its probably relevant or good even to see how the other states are structure to know the differences in their composition. 

If you divide it up I'm worried it will kinda obfuscate / making some details harder to find (searching across files tends to be more annoying than in a file, and looking at things below one another is easier than across 4 or 5 files typically)